### PR TITLE
pipewire: add option to disable systemd support

### PR DESCRIPTION
**Pull request by kampka**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by jansol** [comment]
Note that this also disables soundcard discovery via udev. If udev without systemd is something that is wanted to with pipewire, a separate dependency for just udev should be added. (I don't know how feasible such a setup is nowadays)

**Comment by kampka** [comment]
@jansol valid point, I added `eudev` as an alternative dependency.

File: pkgs/development/libraries/pipewire/default.nix [hunk 1]
```diff
@@ -193,8 +197,11 @@ let
         cp ${buildPackages.pipewire}/nix-support/*.json "$out/nix-support"
       ''}
 
-      moveToOutput "share/systemd/user/pipewire-pulse.*" "$pulse"
-      moveToOutput "lib/systemd/user/pipewire-pulse.*" "$pulse"
+      ${ lib.optionalString (enableSystemd) ''
+        moveToOutput "share/systemd/user/pipewire-pulse.*" "$pulse"
+        moveToOutput "lib/systemd/user/pipewire-pulse.*" "$pulse"
+      '' }
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/pipewire/default.nix:203]:
```suggestion
      ${lib.optionalString enableSystemd ''
        moveToOutput "share/systemd/user/pipewire-pulse.*" "$pulse"
        moveToOutput "lib/systemd/user/pipewire-pulse.*" "$pulse"
      ''}
```

File: pkgs/development/libraries/pipewire/default.nix [hunk 2]
```diff
@@ -127,8 +129,8 @@ let
       vulkan-headers
       vulkan-loader
       webrtc-audio-processing
-      systemd
-    ] ++ lib.optionals gstreamerSupport [ gst_all_1.gst-plugins-base gst_all_1.gstreamer ]
+    ] ++ (if (enableSystemd) then [ systemd ] else [ eudev ])
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/pipewire/default.nix:132]:
```suggestion
    ] ++ (if enableSystemd then [ systemd ] else [ eudev ])
```

File: pkgs/development/libraries/pipewire/default.nix [hunk 3]
```diff
@@ -6,7 +6,9 @@
 , python3
 , meson
 , ninja
+, eudev
 , systemd
+, enableSystemd ? stdenv.isLinux
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/pipewire/default.nix:11]: Does eudev work on something not linux?

Comment (kampka) [reply pkgs/development/libraries/pipewire/default.nix]: Not really. Technically, pipewire supports more platforms than both `systemd` and `eudev`, but that's out of scope for now I guess.
I changed the default to simply `true`, if someone has a need to be more restrictive (eg. `hostPlatform.isStatic` ) we can revisit this I'd say.

File: pkgs/development/libraries/pipewire/default.nix [hunk 4]
```diff
@@ -193,8 +197,11 @@ let
         cp ${buildPackages.pipewire}/nix-support/*.json "$out/nix-support"
       ''}
 
-      moveToOutput "share/systemd/user/pipewire-pulse.*" "$pulse"
-      moveToOutput "lib/systemd/user/pipewire-pulse.*" "$pulse"
+      ${lib.optionalString (enableSystemd) ''
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/pipewire/default.nix:200]:
```suggestion
      ${lib.optionalString enableSystemd ''
```
Those are optional

**Comment by SuperSandro2000** [comment]
also please target staging due to rebuild count

**Comment by kampka** [comment]
> also please target staging due to rebuild count

Yeah, this is unfortunate. The rebuild is only because of the additional whitespace in the `postInstall` hook.

**Comment by SuperSandro2000** [comment]
Please follow the contributing guide the next time when retargeting.

**Pull request merged**